File: 3020-find-the-maximum-number-of-elements-in-subset/3020-find-the-maximum-number-of-elements-in-subset.java
class Solution {
    public int maximumLength(int[] nums) {
       Map<Integer, Integer> freq = new HashMap<>(); 

       for (int num : nums){
        freq.put(num, freq.getOrDefault(num, 0) + 1);
       }

        List<Integer> uniqueNums = new ArrayList<>(freq.keySet());
        uniqueNums.sort(Collections.reverseOrder());
        int maxi = freq.getOrDefault(1,0) % 2 == 1 ? freq.get(1) : freq.getOrDefault(1,0) - 1;

       for (int num: uniqueNums){
        if (freq.get(num) > 0){
            int curr = num;
            freq.put(num, 0);
            int count = 1;

            while (true){
                int temp = (int) Math.sqrt(curr);

                if (temp * temp == curr && freq.getOrDefault(temp,0) >= 2){
                    count += 2;
                    freq.put(temp,0);
                    curr =temp;
                } else {
                    break;
                }
            }

            maxi = Math.max(maxi,count);
        }
       }

       return maxi;
    }
}

File: 3020-find-the-maximum-number-of-elements-in-subset/3020-find-the-maximum-number-of-elements-in-subset.py
class Solution:
    def maximumLength(self, nums: List[int]) -> int:
        nums.sort(reverse = True)
        freq = Counter(nums)
        maxi = freq[1] if freq[1] % 2 == 1 else freq[1] - 1

        for num in nums:
            if num in freq:
                curr = num
                del freq[curr]
                count = 1

                while True:
                    temp = sqrt(curr)

                    if temp == floor(temp) and freq[temp] >= 2:
                        count += 2
                        del freq[temp]
                        curr = temp
                    else:
                        break
                
                maxi = max(maxi,count)

        return maxi